plugins/inputs/kube_state/pod.go

```diff
@@ -0,0 +1,110 @@
+package kube_state
+
+// - resource_requests_cpu_units
+// - resource_limits_cpu_units
+// - resource_requests_memory_bytes
+// - resource_limits_memory_bytes
+// - status_ready
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/ericchiang/k8s/apis/core/v1"
+
+	"github.com/influxdata/telegraf"
+)
+
+func collectPods(ctx context.Context, acc telegraf.Accumulator, ks *KubernetesState) {
+	list, err := ks.client.getPods(ctx)
+	if err != nil {
+		acc.AddError(err)
+		return
+	}
+	for _, p := range list.Items {
+		if err = ks.gatherPod(*p, acc); err != nil {
+			acc.AddError(err)
+			return
+		}
+	}
+}
+
+func (ks *KubernetesState) gatherPod(p v1.Pod, acc telegraf.Accumulator) error {
+	// todo: size is likely wrong
+	if p.Metadata.CreationTimestamp.Size() == 0 {
+		return nil
+	}
+
+	for i, cs := range p.Status.ContainerStatuses {
+		c := p.Spec.Containers[i]
+		gatherPodContainer(*p.Spec.NodeName, p, *cs, *c, acc)
+	}
+
+	for _, c := range p.Status.Conditions {
+		if c.LastTransitionTime.Size() == 0 {
+			continue
+		}
+		gatherPodStatus(p, *c, acc)
+	}
+
+	return nil
+}
+
+func gatherPodContainer(nodeName string, p v1.Pod, cs v1.ContainerStatus, c v1.Container, acc telegraf.Accumulator) {
+	fields := map[string]interface{}{
+		"status_restarts_total":    cs.GetRestartCount(),
+		"status_running":           boolInt(cs.State.Running != nil),
+		"status_terminated":        boolInt(cs.State.Terminated != nil),
+		"status_terminated_reason": cs.State.Terminated.GetReason(),
+	}
+	tags := map[string]string{
+		"container_name": *c.Name,
+		"namespace":      *p.Metadata.Namespace,
+		"node_name":      *p.Spec.NodeName,
+		"pod_name":       *p.Metadata.Name,
+		// "reason":      ,
+	}
+
+	req := c.Resources.Requests
+	lim := c.Resources.Limits
+
+	for resourceName, val := range req {
+		switch resourceName {
+		case "cpu":
+			fields["resource_requests_cpu_units"] = *val.String_
+		case "memory":
+			fields["resource_requests_memory_bytes"] = *val.String_
+		}
+	}
+	for resourceName, val := range lim {
+		switch resourceName {
+		case "cpu":
+			fields["resource_limits_cpu_units"] = *val.String_
+		case "memory":
+			fields["resource_limits_memory_bytes"] = *val.String_
+		}
+	}
+
+	acc.AddFields(podContainerMeasurement, fields, tags)
+}
+
+func gatherPodStatus(p v1.Pod, c v1.PodCondition, acc telegraf.Accumulator) {
+	tags := map[string]string{
+		"namespace": p.Metadata.GetNamespace(),
+		"pod_name":  p.Metadata.GetName(),
+		"node_name": p.Spec.GetNodeName(),
+		"reason":    p.Status.GetReason(),
+	}
+
+	fields := make(map[string]interface{})
+
+	switch strings.ToLower(*c.Type) {
+	case "ready":
+		fields["ready"] = "true"
+	default:
+		fields["ready"] = "false"
```

Comment:
they do have more conditions, they get added to the pod as they happen, `ready` is the only one requested. others include "PodScheduled" and "Initialized". a pod can have one or all of the conditions

Reply:
c.Type is a *string, how do they set it to have more than one condition?

Reply:
Hmm, okay I have a few ideas for this, but I'm not 100% how this data changes over time.  Does it start with 0 or 1 condition and then add them over time?  Maybe it always has all conditions but status is false?  Can you check and show how this data changes over time as a pod comes up.

Reply:
When you create a pod, it starts with a single condition type `PodScheduled` with a status `True`. As it comes online `Ready` and `Initialized` types are appended into the condition list. `Initialized` has a status of `True` (within one second of pod create), and `Ready`'s state transitions to `True` once the pod is online. I'm not certain when/if the other two condition types (`Unschedulable` and `ContainersReady`) come into play. It seems that an event is not triggered if/when they do (if one is `watch`ing, rather than `list`ing)

Reply:
This feels like another thing, similar to the configmap, that almost seems a better fit for event style metrics...

> It seems that an event is not triggered if/when they do

Are you saying there are never events triggered for pod-condition changes, or only you don't see them for Unschedulable and ContainersReady?

Reply:
In my limited tests, (working on 5281), I didn't see k8s fire an event for "unschedulable" or "containersReady" while `watch`ing pods. It might happen though.

Reply:
But you did see them for Initialized and Ready?